PubMed XML.js

```diff
@@ -0,0 +1,566 @@
+{
+	"translatorID": "fcf41bed-0cbc-3704-85c7-8062a0068a7a",
+	"label": "PubMed XML",
+	"creator": "Simon Kornblith, Michael Berkowitz, Avram Lyon, and Rintze Zelle",
+	"target": "xml",
+	"minVersion": "2.1.9",
+	"maxVersion": "",
+	"priority": 100,
+	"configOptions": {
+		"dataMode": "xml/dom"
+	},
+	"inRepository": true,
+	"translatorType": 1,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2015-03-21 12:27:58"
+}
+
+/*******************************
+ * Import translator functions *
+ *******************************/
+
+function detectImport() {
+	Zotero.debug("Detecting Pubmed content....");
+	// Look for the PubmedArticle tag in the first 1000 characters
+	var text = Zotero.read(1000);
+	if (text.indexOf("<PubmedArticle>") != -1) return "journalArticle";
+	if (text.indexOf("<PubmedBookArticle>") != -1) return "book";
+	return false;
+}
+
+function processAuthors(newItem, authorsLists) {
+	for(var j=0, m=authorsLists.length; j<m; j++) {
+		//default to 'author' unless it's 'editor'
+		var type = "author";
+		if(authorsLists[j].hasAttribute('Type')
+			&& authorsLists[j].getAttribute('Type') === "editors") {
+			type = "editor";
+		}
+	
+		var authors = ZU.xpath(authorsLists[j], 'Author');
+	
+		for(var k=0, l=authors.length; k<l; k++) {
+			var author = authors[k];
+			var lastName = ZU.xpathText(author, 'LastName');
+			var firstName = ZU.xpathText(author, 'FirstName');
+			if(!firstName) {
+				firstName = ZU.xpathText(author, 'ForeName');
+			}
+	
+			var suffix = ZU.xpathText(author, 'Suffix');
+			if(suffix && firstName) {
+				firstName += ", " + suffix
+			}
+	
+			if(firstName || lastName) {
+				var creator = ZU.cleanAuthor(lastName + ', ' + firstName, type, true);
+				if(creator.lastName.toUpperCase() == creator.lastName) {
+					creator.lastName = ZU.capitalizeTitle(creator.lastName, true);
+				}
+				if(creator.firstName.toUpperCase() == creator.firstName) {
+					creator.firstName = ZU.capitalizeTitle(creator.firstName, true);
+				}
+				newItem.creators.push(creator);
+			} else if(lastName = ZU.xpathText(author, 'CollectiveName')) {
+				//corporate author
+				newItem.creators.push({
+					creatorType: type,
+					lastName: lastName,
+					fieldMode: 1
+				});
+			}
+		}
+	}
+}
+
+function doImport() {
+
+	var doc = Zotero.getXML();
+
+	if ( ZU.xpathText(doc, '//*').length < 200 ) {
+		throw("No Pubmed Data found - Most likely eutils is temporarily down")
+	}
```

Comment:
Not the right place to test this (we can do it in the PubMed translator). If PubMed server fails to serve the XML, it will just fail to parse at the `Zotero.getXML()` call.